### ISSUE TITLE
Where exactly are the KL losses used after the forward pass?

### ISSUE THREAD

**Issue by nbro** [tensorflow/probability#885]
I've noticed that the KL part of the loss is added to the list [`self._losses` of the `Layer` class](https://github.com/keras-team/keras/blob/master/keras/engine/base_layer.py#L117) when [`self.add_loss` is called from the `call` method](https://github.com/tensorflow/probability/blob/v0.9.0/tensorflow_probability/python/layers/dense_variational.py#L308) of the `DenseVariational` (i.e. during the forward pass).

But how is this list `self._losses` (or the method `losses` of the same `Layer` class) treated during training? Where is it called from during training? I would like to know how exactly these losses are combined with the loss that you specify in the `fit` method.

**Comment by nbro** [comment]
See https://stackoverflow.com/q/61314548/3924118.
